usecases/usecases.go

```diff
@@ -0,0 +1,93 @@
+package usecases
+
+type ProgrammingLanguage string
+type UserId string
+type SnippetId string
+type CommentId string
+type Token string
+
+type User struct {
+	Id       UserId // Unique identifier, persists through username changes
+	Username string // Visible username, can be changed
+}
+
+type Rating struct {
+	Likes    int
```

Comment:
Implementing likes and dislikes as simple counts will make it impossible to prevent multiple updates from a given user.

Reply:
Is it really necessary to do in usecases? This seems like a persistence layer problem, though there definitely should be a way to retrieve user's votes.